docs/gc.md

```diff
@@ -95,23 +98,32 @@ This mode can be enabled with hardware-based support and software-based support:
 
-Regions impose a maximum object size. Objects are always allocated within the bounds of a single region and are never permitted to span regions. The region size is always a power of two; for example, 512 KB, 1 MB, and so on (where KB is 2<sup>10</sup> bytes and MB is 2<sup>20</sup> bytes). The region size is selected at startup based on the maximum heap size. The collector chooses the smallest power of two which will result in less than 2048 regions, with a minimum region size of 512 KB. Except for small heaps (less than about 512 MB) the VM aims to have between 1024 and 2047 regions. 
+Regions impose a maximum object size. Objects are always allocated within the bounds of a single region and are never permitted to span regions. The region size is always a power of two; for example, 512 KB, 1 MB, and so on (where KB is 2<sup>10</sup> bytes and MB is 2<sup>20</sup> bytes). The region size is selected at startup based on the maximum heap size. The collector chooses the smallest power of two which will result in less than 2048 regions, with a minimum region size of 512 KB. Except for small heaps (less than about 512 MB) the VM aims to have between 1024 and 2047 regions.
 
-For further information see [Balanced Garbage Collection policy](https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.vm.80.doc/docs/mm_gc_balanced.html).
+Object ages are tracked for each region with a maximum of 24 possible generations. The following diagram illustrates the structure of the object heap:
+
+![The diagram is explained in the surrounding text](./cr/balanced_regions.png "Region structure and characteristics found in the object heap")
+
+ An age 0 region, known as the *eden* space, contains the newest objects allocated. Regions of intermediate age serve as survivor regions to receive live objects copied forwarded from younger regions. When the eden space is full, a partial garbage collection (PGC) runs to reclaim space by using either a *copy-forward* or *mark-compact* operation. The PGC is a stop-the-world operation, after which new regions are allocated to create a new eden space. A PGC always includes the eden space, but might include older regions. Objects within collectible regions of intermediate age N are evacuated to survivor regions of age N+1. Regions of age 24 are never included in PGC collection sets but might be included for compaction in global GC (GGC) cycles, which mark and sweep the whole heap and compact regions that need defragmentation.
```

Comment:
we may want to emphasize that copy-forward is preferred over mark-compact (since the former is a single-pass GC and causes shorter pauses than the latter which is a 2-pass GC)

.... reclaim space by typically using a *copy-forward* operation, although in some specific cases (for example when there is not enough free survivor regions) *mark-compact* operation might be used.

Reply:
Updated to: 

"When the eden space is full, a partial garbage collection (PGC) runs to reclaim space, typically by using a *copy-forward* operation. In specific cases, a *mark-compact* operation might be used, for example, when there are not enough free survivor regions available."